crates/next-api/src/nft_json.rs

```diff
@@ -0,0 +1,191 @@
+use anyhow::{bail, Result};
+use serde_json::json;
+use turbo_tasks::{RcStr, ResolvedVc, ValueToString, Vc};
+use turbo_tasks_fs::{DiskFileSystem, File, FileSystem, FileSystemPath, VirtualFileSystem};
+use turbopack_core::{
+    asset::{Asset, AssetContent},
+    ident::AssetIdent,
+    output::OutputAsset,
+    reference::all_assets_from_entries,
+};
+
+/// A json file that produces references to all files that are needed by the given module
+/// at runtime. This will include, for example, node native modules, unanalyzable packages,
+/// client side chunks, etc.
+///
+/// With this file, users can determine the minimum set of files that are needed alongside
+/// their bundle.
+#[turbo_tasks::value(shared)]
+pub struct NftJsonAsset {
+    /// The chunk for which the asset is being generated
+    chunk: Vc<Box<dyn OutputAsset>>,
+    output_fs: Vc<DiskFileSystem>,
+    project_fs: Vc<DiskFileSystem>,
```

Comment:
Avoid using a concrete filesystem implementation. Better make this filesystem Independent and only use project and output root paths as `output_root: ResolvedVc<FileSystemPath>` and project_root